iroh-net/src/magicsock.rs

```diff
@@ -1282,6 +1282,12 @@ impl MagicSock {
                 return Ok(current_value.clone().into_iter().collect());
             }
         }
+
+        self.local_endpoints_change().await
+    }
+
+    /// Waits for local endpoints to change and returns the new ones.
+    pub async fn local_endpoints_change(&self) -> Result<Vec<config::Endpoint>> {
```

Comment:
I can't help but bikeshed, something about `_change` doesn't feel right.  It's almost like you want to return a `Watchable<Vec<config::Endpoint>>`, but that would expose a foreign crate on the API which is no good.

Maybe just `local_endpoints_updated()` is better already?  I think the tense of `change` is also bothering me.

Reply:
both `changed` and `updated` sound to me like it should return a boolean to me

Reply:
maybe `next_local_endpoints`, matching the stream api, as it behaves similarly like that

Reply:
`next_local_endpoints` gives me the expectation of an `Option`... naming is hard.  maybe putting the verb at the front is better? `updated_local_endpoints`

let's not make this a blocker, it's not worth it.  and whatever we'll come up with now probably won't be great anyway.  maybe someone will realise a great name while using it and then we can change the name as long as we're not 1.0.